### PR TITLE
Bugfix/ins 3337

### PR DESCRIPTION
**Pull request by RockMother**
Renamed propertyMetadata into dictionaries.
Refactored property configuration.

### CODE REVIEW

**Review by aehrenthal — APPROVED** [review]

File: sell/sell-api.yaml [hunk 1]
```diff
@@ -5560,10 +5515,16 @@ components:
...
-              additionalProperties: true
-              description: Additional configuration of the property.
+            dataTable:
+              type: boolean
+            details:
+              type: boolean
+            bulkUpdate:
+              type: boolean
+            template:
+              type: boolean
```
Comment (olyavertwist) [inline sell/sell-api.yaml:5525]: description: The true/false parameter indicating if this property supports templates.

File: sell/sell-api.yaml [hunk 2]
```diff
@@ -5560,10 +5515,16 @@ components:
...
+            filtering:
+              type: boolean
```
Comment (olyavertwist) [inline sell/sell-api.yaml:5527]: description: The true/false parameter indicating if this property supports filtering.

File: sell/sell-api.yaml [hunk 3]
```diff
@@ -106,9 +106,6 @@ tags:
   - name: Pipeline Statistics
     description: |
       Analyze your sales progress with pipeline statistics.
-  - name: Predefined Metadata
```
Comment (olyavertwist) [inline sell/sell-api.yaml:109]: We need to add a Dictionaries tag to Tags.

**Review by olyavertwist — CHANGES_REQUESTED** [review]
Added comments about descriptions.

**Review by Zemidar — APPROVED** [review]

File: sell/sell-api.yaml [hunk 4]
```diff
@@ -2531,66 +2528,44 @@ paths:
...
-  /predefinedMetadataGroups:
+  /dictionaries:
     get:
       tags:
-        - Predefined Metadata
-      summary: List metadata groups
-      description: Use this endpoint to get a list of dictionaries.
-      operationId: getAllPredefinedMetadataGroups
+        - Dictionaries
+      summary: Retrieve a list of available dictionaries
```
[the remark this answers is not in the document]
Comment (RockMother) [reply sell/sell-api.yaml:2535]: Done

File: sell/sell-api.yaml [hunk 5]
```diff
@@ -2531,66 +2528,44 @@ paths:
...
+      description: Use this endpoint to get list of all avaliable dictionaries.
```
[the remark this answers is not in the document]
Comment (RockMother) [reply sell/sell-api.yaml:2536]: Done

File: sell/sell-api.yaml [hunk 6]
```diff
@@ -2531,66 +2528,44 @@ paths:
...
+      operationId: getDictionaries
       responses:
         200:
-          description: The list of predefined metadata groups has been retrieved. Returns predefined metadata groups.
+          description: The dictionary has been retrieved. Returns dictionary items.
```
[the remark this answers is not in the document]
Comment (RockMother) [reply sell/sell-api.yaml:2540]: Done

File: sell/sell-api.yaml [hunk 7]
```diff
@@ -5502,12 +5455,14 @@ components:
         value:
           type: string
           description: The value of the option. In most cases, the value is the same as the label.
+        group:
+          type: string
       description: The dictionary entry.
-    PredefinedMetadataOptionList:
+    Dictionary:
       type: array
```
[the remark this answers is not in the document]
Comment (olyavertwist) [reply sell/sell-api.yaml:5462]: dictionaries/id is removed, so no need to add id to the dictionary object

**Review by olyavertwist — CHANGES_REQUESTED** [review]
Just one tiny fix - a typo (marka >> marks)

File: sell/sell-api.yaml [hunk 8]
```diff
@@ -5560,10 +5515,21 @@ components:
...
-            configuration:
-              type: object
-              additionalProperties: true
-              description: Additional configuration of the property.
+            dataGrid:
+              type: boolean
+              description: The parameter marks if this property should be displayed in the table view (e.g., on the Contacts page, the Companies page).
+            detailsPage:
+              type: boolean
+              description: The parameter marka if this property should be displayed on the object card (e.g., when a user drills down to a company).
```
[the remark this answers is not in the document]
Comment (RockMother) [reply sell/sell-api.yaml:5523]: Done

**Review by olyavertwist — APPROVED** [review]

**Pull request merged**